core/io/resource_loader.h

```diff
@@ -230,7 +230,13 @@ class ResourceLoader {
 	// Loaders can safely use this regardless which thread they are running on.
 	static void notify_dependency_error(const String &p_path, const String &p_dependency, const String &p_type) {
 		if (dep_err_notify) {
-			callable_mp_static(dep_err_notify).bind(p_path, p_dependency, p_type).call_deferred();
+			if (Thread::get_caller_id() == Thread::get_main_id()) {
+				Callable::CallError ce;
+				Variant ret;
+				callable_mp_static(dep_err_notify).bind(p_path, p_dependency, p_type).callp(nullptr, 0, ret, ce);
+			} else {
+				callable_mp_static(dep_err_notify).bind(p_path, p_dependency, p_type).call_deferred();
+			}
```

Comment:
The main-thread case can just be "natural":

```suggestion
			if (Thread::get_caller_id() == Thread::get_main_id()) {
				dep_err_notify(p_path, p_dependency, p_type);
			} else {
				callable_mp_static(dep_err_notify).bind(p_path, p_dependency, p_type).call_deferred();
			}
```